### PR TITLE
quiet down redundant warnings for binary targets with multiple info.plist

### PR DESCRIPTION
**Pull request by tomerd**
motivation: better ux

changes:
* change un-parsable info.plist warning to debug since its a valid use case and the warning is intended mainly for deubg purposes
* add informational diagnostics with the details of the found  the binary artifact
* formatting

rdar://107669671

### CODE REVIEW

File: Sources/Workspace/Workspace+BinaryArtifacts.swift [hunk 1]
```diff
@@ -521,7 +692,7 @@ extension Workspace.BinaryArtifactsManager {
                 _ = try decoder.decode(XCFrameworkMetadata.self, from: fileSystem.readFileContents(infoPlist))
                 return .xcframework
             } catch {
-                observabilityScope.emit(warning: "info.plist found in '\(path)' but failed to parse: \(error)")
+                observabilityScope.emit(debug: "info.plist found in '\(path)' but failed to parse: \(error)")
```
Comment (tomerd) [inline Sources/Workspace/Workspace+BinaryArtifacts.swift:695]: the actual change

File: Sources/Workspace/Workspace+BinaryArtifacts.swift [hunk 2]
```diff
@@ -470,49 +614,76 @@ extension Workspace.BinaryArtifactsManager {
...
             // multiple ones, return the last one to preserve old behavior
-            observabilityScope.emit(warning: "multiple potential binary artifacts found: '\(binaryArtifacts.map{ $0.0.description }.joined(separator: "', '"))', using the one in '\(binaryArtifact.0)'")
+            observabilityScope
+                .emit(
+                    warning: "multiple potential binary artifacts found: '\(binaryArtifacts.map(\.0.description).joined(separator: "', '"))', using the one in '\(binaryArtifact.0)'"
+                )
             return binaryArtifact
         } else if let binaryArtifact = binaryArtifacts.first {
             // single one
+            observabilityScope.emit(info: "found binary artifact: '\(binaryArtifact)'")
```
Comment (tomerd) [inline Sources/Workspace/Workspace+BinaryArtifacts.swift:636]: the actual change

**Comment by tomerd** [comment]
@swift-ci smoke test

**Review by neonichu — APPROVED** [review]

**Comment by tomerd** [comment]
@swift-ci smoke test macOS

**Comment by tomerd** [comment]
@swift-ci smoke test macOS

**Review by MaxDesiatov — APPROVED** [review]

**Pull request merged**